versioned/tiered/rocksdb/src/main/java/org/projectnessie/versioned/rocksdb/RocksRef.java

```diff
@@ -0,0 +1,362 @@
+        throw new IllegalArgumentException(String.format("Unknown identifier name [%s].", identifier));
+      }
+    }
+  }
+
+  private Type type = Type.INIT;
+
+  @Override
+  public boolean evaluateFunction(Function function) {
+    if (type == Type.BRANCH) {
+      return evaluateBranch(function);
+    } else if (type == Type.TAG) {
+      return evaluateTag(function);
+    }
+    return false;
+  }
+
+  /**
+   * Evaluates that this branch meets the condition.
+   *
+   * @param function the function that is tested against the nameSegment
+   * @return true if this branch meets the condition
+   */
+  private boolean evaluateBranch(Function function) {
+    final String segment = function.getRootPathAsNameSegment().getName();
+
+    switch (segment) {
+      case ID:
+        return idEvaluates(function);
+      case TYPE:
+        return (function.isRootNameSegmentChildlessAndEquals()
+          && type.equals(Type.getType(function.getValue().getString())));
+      case NAME:
+        return (function.isRootNameSegmentChildlessAndEquals()
+          && name.equals(function.getValue().getString()));
+      case CHILDREN:
+        return evaluateStream(function, children);
+      case METADATA:
+        return (function.isRootNameSegmentChildlessAndEquals()
+          && metadata.toEntity().equals(function.getValue()));
+      case COMMITS:
+        // TODO: refactor once jdbc-store Store changes are available.
+        switch (function.getOperator()) {
+          case SIZE:
+            return (!function.getRootPathAsNameSegment().getChild().isPresent()
+              && commits.size() == function.getValue().getNumber());
+          default:
+            return false;
+        }
+      default:
+        return false;
+    }
+  }
+
+  /**
+   * Evaluates that this tag meets the condition.
+   *
+   * @param function the function that is tested against the nameSegment
+   * @return true if this tag meets the condition
+   */
+  private boolean evaluateTag(Function function) {
+    final String segment = function.getRootPathAsNameSegment().getName();
+    switch (segment) {
+      case ID:
+        return idEvaluates(function);
+      case TYPE:
+        return (function.isRootNameSegmentChildlessAndEquals()
+          && type.equals(Type.getType(function.getValue().getString())));
+      case NAME:
+        return (function.getOperator().equals(Function.EQUALS)
+          && name.equals(function.getValue().getString()));
+      case COMMIT:
+        return (function.getOperator().equals(Function.EQUALS)
+          && commit.toEntity().equals(function.getValue()));
+      default:
+        return false;
+    }
+  }
+
+  @Override
+  public Ref name(String name) {
+    this.name = name;
+    return this;
+  }
+
+  @Override
+  public Tag tag() {
+    if (type != Type.INIT) {
+      throw new IllegalStateException("branch()/tag() has already been called");
+    }
+    type = Type.TAG;
+    // TODO: Do we need to serialize ref type?
```

Comment:
@stevelorddremio should this still be here?

Reply:
No.